memgpt/server/rest_api/app.py

```diff
@@ -0,0 +1,169 @@
+from fastapi import FastAPI
+from starlette.middleware.cors import CORSMiddleware
+
+from memgpt.server.constants import REST_DEFAULT_PORT
+
+# NOTE(charles): these are extra routes that are not part of v1 but we still need to mount to pass tests
+from memgpt.server.rest_api.auth.index import (
+    setup_auth_router,  # TODO: probably remove right?
+)
+from memgpt.server.rest_api.interface import StreamingServerInterface
+from memgpt.server.rest_api.routers.openai.assistants.assistants import (
+    router as openai_assistants_router,
+)
+from memgpt.server.rest_api.routers.openai.assistants.threads import (
+    router as openai_threads_router,
+)
+from memgpt.server.rest_api.routers.openai.chat_completions.chat_completions import (
+    router as openai_chat_completions_router,
+)
+
+# from memgpt.orm.utilities import get_db_session  # TODO(ethan) reenable once we merge ORM
+from memgpt.server.rest_api.routers.v1 import ROUTERS as v1_routes
+from memgpt.server.rest_api.routers.v1.users import (
+    router as users_router,  # TODO: decide on admin
+)
+from memgpt.server.rest_api.static_files import mount_static_files
+from memgpt.server.server import SyncServer
+from memgpt.settings import settings
+
+# TODO(ethan)
+# NOTE(charles): @ethan I had to add this to get the global as the bottom to work
+interface: StreamingServerInterface = StreamingServerInterface
+server: SyncServer = SyncServer(default_interface_factory=lambda: interface())
+
+# TODO(ethan): eventuall remove
+if password := settings.server_pass:
+    # if the pass was specified in the environment, use it
+    print(f"Using existing admin server password from environment.")
+else:
+    # Autogenerate a password for this session and dump it to stdout
+    password = secrets.token_urlsafe(16)
+    typer.secho(f"Generated admin server password for this session: {password}", fg=typer.colors.GREEN)
+
+
+ADMIN_PREFIX = "/v1/admin"
+API_PREFIX = "/v1"
+OPENAI_API_PREFIX = "/openai"
+
+
+def create_application() -> "FastAPI":
+    """the application start routine"""
+
+    app = FastAPI(
+        swagger_ui_parameters={"docExpansion": "none"},
+        # openapi_tags=TAGS_METADATA,
+        title="MemGPT",
+        summary="Create LLM agents with long-term memory and custom tools 📚🦙",
+        version="1.0.0",  # TODO wire this up to the version in the package
+    )
+    app.add_middleware(
+        CORSMiddleware,
+        allow_origins=settings.cors_origins,
+        allow_credentials=True,
+        allow_methods=["*"],
+        allow_headers=["*"],
+    )
+
+    for route in v1_routes:
+        app.include_router(route, prefix=API_PREFIX)
+        # this gives undocumented routes for "latest" and bare api calls.
+        # we should always tie this to the newest version of the api.
+        app.include_router(route, prefix="", include_in_schema=False)
+        app.include_router(route, prefix="/latest", include_in_schema=False)
+
+    # NOTE: ethan these are the extra routes
```

Comment:
Note @norton120 I had to add these extra routes, some for admin/auth and others to wire in the OpenAI assistants + chatcompletions routes (not sure how / if they were being wired in before with the original `app.py`?

Reply:
k we can revisit in a follow-up PR, ideally we'll roll them all into the same iteration so if we add a `V1.5` API it is just changing one value in here - but we can do that in a follow-up PR